csi/controller.go

```diff
@@ -266,6 +276,17 @@ func (s *OsdCsiServer) CreateVolume(
 		return nil, status.Error(codes.InvalidArgument, e)
 	}
 
+	// Get parent ID from request: snapshot or volume
+	if req.GetVolumeContentSource() != nil {
```

Comment:
A test in csi-test was added to check that this is implemented. That was failing, so I added the implementation in this PR.